lib/components/src/controls/types.ts

```diff
@@ -9,6 +9,7 @@ export interface ControlProps<T> {
   onChange: (value: T) => T | void;
   onFocus?: (evt: any) => void;
   onBlur?: (evt: any) => void;
+  readOnly?: boolean;
```

Comment:
This doesn't look right to me, it fixes TS types in the various control option components, but I'm not sure it's correct.

Also, if it's correct, the `disable` property should be added there, too.